src/main/java/net/minecraftforge/event/datafix/ConfigureDatafixSchemaEvent.java

```diff
@@ -0,0 +1,179 @@
+package net.minecraftforge.event.datafix;
+
+import com.google.common.collect.Maps;
+import com.mojang.datafixers.DSL;
+import com.mojang.datafixers.schemas.Schema;
+import com.mojang.datafixers.types.templates.TypeTemplate;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.datafix.fixes.References;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.event.IModBusEvent;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Event fired in parallel on the mod event bus to allow for the modification
+ * of a DFU Schemas type signature.
+ *
+ * If used to register a new entity or block entity to a particular version,
+ * then this also will need to be registered to all newer versions to persist properly!
+ * As opposed to vanilla which automatically transfers the entities and block entities from older
+ * schemas to newer ones, this event does not do that and the modder is on his own to implement this properly.
+ *
+ * We suggest instead of checking for version equality to check for a greater or equal version.
+ */
+public class ConfigureDatafixSchemaEvent extends Event implements IModBusEvent
```

Comment:
Given how DataFixers are named elsewhere I believe this should be called `ConfigureDataFixSchemaEvent ` (basically make the `f` capital `F`)